File: 1967. Number of Strings That Appear as Substrings in Word.py
from typing import List

class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        ans=0

        for pattern in patterns:
            n=len(pattern)

            for j in range(len(word)):
                if word[j] == pattern[0]:
                    if j + n <= len(word) and word[j:j + n] == pattern:
                        ans += 1
                        break
        
        return ans

File: 1967. Number of Strings That Appear as Substrings in Word.cpp
class Solution {
public:
    int numOfStrings(vector<string>& patterns, string word) {
        int ans = 0;

        for(int i = 0; i < patterns.size(); i++)
        {
            int len = patterns[i].size();

            for(int j = 0; j < word.size(); j++)
            {
                if(word[j] == patterns[i][0])
                {
                    if(word.substr(j, len) == patterns[i])
                    {
                        ans++;
                        break;
                    }
                }
            }
        }

        return ans;
    }
};